reliability-engineering/terraform/modules/concourse-secrets-admin/iam.tf

```diff
@@ -0,0 +1,82 @@
+resource "aws_iam_role" "concourse_secrets_admin" {
+  name = "${var.deployment}-${var.concourse_team_name}-concourse-secrets-admin"
+
+  assume_role_policy = jsonencode({
+    Version = "2012-10-17"
+    Statement = [
+      {
+        Effect = "Allow"
+        Principal = {
+          Federated = module.concourse_base.aws_iam_openid_connect_provider_arn
+        }
+        Action = "sts:AssumeRoleWithWebIdentity"
+        Condition = {
+          StringEquals = {
+            "${var.oidc_host_path}:aud" = var.github_oauth_client_id
+            "aws:RequestTag/t${var.trusted_github_team_id}" = "t"
+          }
+        }
+      }, {
+        Sid = "AllowPassSessionTagsAndTransitive"
+        Effect = "Allow"
+        Action = "sts:TagSession"
+        Principal = {
+          Federated = module.concourse_base.aws_iam_openid_connect_provider_arn
+        }
+      }
+    ]
+  }
+}
+
+resource "aws_iam_role_policy" "concourse_secrets_admin" {
+  name = "${var.deployment}-${var.concourse_team_name}-concourse-secrets-admin-policy"
+  role = aws_iam_role.concourse_secrets_admin.id
+
+  policy = jsonencode({
+    Version = "2012-10-17"
+    Statement = [
+      {
+        Action = [
+          "ssm:GetParameter",
+          "ssm:GetParameters",
+          "ssm:GetParametersByPath",
+        ]
+        Effect = "Allow"
+        Resource = [
+          "arn:aws:ssm:eu-west-2:${data.aws_caller_identity.account.account_id}:parameter/${var.deployment}/concourse/worker/${var.concourse_team_name}/*"
```

Comment:
why allow this?

Reply:
This policy was adapted from the worker one, and this bit looked plausible to keep in at the time. Looking at it again now, it looks like it's got something to do with the internal operation of the workers, rather than anything users need to interact with.

I've removed it.